From 134b1251174395e75f638d53e5449815dc965e8b Mon Sep 17 00:00:00 2001
From: Evan Severson <evanseverson@google.com>
Date: Tue, 30 Nov 2021 18:19:18 -0800
Subject: [PATCH] [BACKPORT] Hide overlays on ReviewPermissionsAtivity

Test: atest PermissionReviewTapjackingTest
Bug: 176094367
Merged-In: I9f263b947853e14d081a73ce907917e9326b6ef7
Change-Id: I9f263b947853e14d081a73ce907917e9326b6ef7
(cherry picked from commit 725244f010c9c5ed5b169c2ec00600864fce38ab)
Merged-In: I9f263b947853e14d081a73ce907917e9326b6ef7
---
 .../permission/ui/ReviewPermissionsActivity.java              | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/com/android/packageinstaller/permission/ui/ReviewPermissionsActivity.java b/src/com/android/packageinstaller/permission/ui/ReviewPermissionsActivity.java
index 6bc251aed..829e81856 100644
--- a/src/com/android/packageinstaller/permission/ui/ReviewPermissionsActivity.java
+++ b/src/com/android/packageinstaller/permission/ui/ReviewPermissionsActivity.java
@@ -39,6 +39,7 @@
 import android.text.style.ForegroundColorSpan;
 import android.util.TypedValue;
 import android.view.View;
+import android.view.WindowManager;
 import android.widget.Button;
 import android.widget.ImageView;
 import android.widget.TextView;
@@ -57,6 +58,9 @@ public final class ReviewPermissionsActivity extends Activity
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
+        getWindow().addPrivateFlags(
+                WindowManager.LayoutParams.PRIVATE_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
+
         PackageInfo packageInfo = getTargetPackageInfo();
         if (packageInfo == null) {
             finish();
